### ISSUE TITLE
Question: what about emulating ping option

### ISSUE THREAD

**Issue by opaquewarellc** [dburckh/AndroidPing#1]
Can AndroidPing emulate ping with option parameter? e.g. 'ping -a ip' to get the host name. 
Thx,

**Comment by dburckh** [comment]
No, the lib just does basic ping functionality.  The functionality you are
looking for would be more like nslookup.

Dustin

**Comment by hardikbhalodi** [comment]
How to make traceroute functionality by increasing TTL, i was able to achieve with native ping CLI command but don't know how to do with this lib. i know OS level API are capable for that but don't know how to implement it. I have seen one example here which using same API as in this lib but coded in CPP. Any help much appreciated.
https://github.com/steamclock/internetmap/blob/master/Android-NDK/tracepath.cpp

**Comment by dburckh** [comment]
In theory you should be able to manipulate the TTL with Os.setsockoptInt
and OsConstants.IP_TTL

Dustin

**Comment by hardikbhalodi** [comment]
@dburckh Thank you for reply, do you have any example or could you make demo lib for this? Also  not sure why this app lib throwing exception in Android LOLLIPOP 5 and 5.1 for other version its working, i Tested in emulator.

**Comment by dburckh** [comment]
I've never actually seen this done, I just believe it should be possible
based on Wikipedia.  It going to take some work on your end.

Dustin
